File: 350_lambda_expressions_in_java/counter.js
function makeCounter(init){
    return () => {
        init++;
        return init-1;
    }
}

c1 = makeCounter(1);
c2 = makeCounter(1);

console.log(c1());
console.log(c1());
console.log(c2());
console.log(c2());

File: 350_lambda_expressions_in_java/Counter.cs
using System;
public class Counter {
    static Func<int> makeCounter(int init){
        return () => {
            init++;
            return init-1;
        };
    }
    public static void Main(){
        Func<int> c1, c2;
        c1 = makeCounter(1); c2 = makeCounter(1);
        Console.WriteLine(c1()); Console.WriteLine(c1());
        Console.WriteLine(c2()); Console.WriteLine(c2());
    }
}